exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/ContextFunctions.java

```diff
@@ -64,17 +65,45 @@ public void eval() {
     @Inject DrillBuf buffer;
     @Workspace int currentSchemaBytesLength;
 
+    @Override
     public void setup() {
       final byte[] currentSchemaBytes = contextInfo.getCurrentDefaultSchema().getBytes();
       buffer = buffer.reallocIfNeeded(currentSchemaBytes.length);
       currentSchemaBytesLength= currentSchemaBytes.length;
       buffer.setBytes(0, currentSchemaBytes);
     }
 
+    @Override
     public void eval() {
       out.start = 0;
       out.end = currentSchemaBytesLength;
       out.buffer = buffer;
     }
   }
+
+  /**
+   * Implement "session_id" function. Returns the unique id of the current session.
+   */
+  @FunctionTemplate(name = "session_id", scope = FunctionTemplate.FunctionScope.SIMPLE, isNiladic = true)
```

Comment:
Please exclude `timeofday, now, statement_timestamp, transaction_timestamp, unix_timestamp`. Just leave those that already behave like niladic functions (just execute function in Drill and see if it requires parenthesis or not, if not, then it's niladic :)). We don't won't to change existing behavior not to break queries.